content/en/docs/tasks/manage-kubernetes-objects/storage-version-migration.md

```diff
@@ -0,0 +1,312 @@
+  ```yaml
+  kind: EncryptionConfiguration
+  apiVersion: apiserver.config.k8s.io/v1
+  resources:
+  - resources:
+    - secrets
+      providers:
+    - aescbc:
+      keys:
+      - name: key2
+        secret: c2VjcmV0IGlzIHNlY3VyZSwgaXMgaXQ/
+    - aescbc:
+      keys:
+      - name: key1
+        secret: c2VjcmV0IGlzIHNlY3VyZQ==
+  ```
+- To ensure that previously created secret `my-secert` is re-encrypted
+with new key `key2`, you will use _Storage Version Migration_.
+- Create `StorageVersionMigration` resource file with the name `migrate-secret.yaml`
+as follows:
+  ```yaml
+  kind: StorageVersionMigration
+  apiVersion: storagemigration.k8s.io/v1alpha1
+  metadata:
+    name: secrets-migration
+  spec:
+    resource:
+      group: ""
+      version: v1
+      resource: secrets
+  ```
+  Create the resource using _kubectl_ as follows:
+  ```shell
+  kubectl apply -f migrate-secret.yaml
+  ```
+- Monitor migration of Secrets using status. Successful migration should have
+_Succeeded_ condition set to _true_ in the status field. Get the migration resource
+as follows:
+  ```shell
+  kubectl get storageversionmigration.storagemigration.k8s.io/secrets-migration -o yaml
+  ```
+  This will produce an output similar to:
+  ```yaml
+  kind: StorageVersionMigration
+  apiVersion: storagemigration.k8s.io/v1alpha1
+  metadata:
+    name: secrets-migration
+    uid: 628f6922-a9cb-4514-b076-12d3c178967c
+    resourceVersion: '90'
+    creationTimestamp: '2024-03-12T20:29:45Z'
+  spec:
+    resource:
+      group: ""
+      version: v1
+      resource: secrets
+  status:
+    conditions:
+    - type: Running
+      status: 'False'
+      lastUpdateTime: '2024-03-12T20:29:46Z'
+      reason: StorageVersionMigrationInProgress
+    - type: Succeeded
+      status: 'True'
+      lastUpdateTime: '2024-03-12T20:29:46Z'
+      reason: StorageVersionMigrationSucceeded
+    resourceVersion: '84'
+  ```
+- [Verify](/docs/tasks/administer-cluster/kms-provider/#verifying-that-the-data-is-encrypted)
+  the stored secret is now prefixed with `k8s:enc:aescbc:v1:key2`.
+
+## Update the preferred storage schema of the resource while moving from _v1_ to _v2_
+Consider a scenario where a {{< glossary_tooltip term_id="CustomResourceDefinition" text="CustomResourceDefinition" >}}
+(CRD) is created to serve custom resources (CRs) and is set as the preferred storage schema. When it's time
+to introduce v2 of the CRD, it can be added for serving only with a conversion
+webhook. This enables a smoother transition where users can create CRs using
+either the v1 or v2 schema, with the webhook in place to perform the necessary
+schema conversion between them. Before setting v2 as the preferred storage schema
+version, it's important to ensure that all existing CRs stored as v1 are migrated to v2.
+This migration can be achieved through _Storage Version Migration_ to migrate all CRs from v1 to v2.
+
+- Create a manifest for the CRD, named `test-crd.yaml`, as follows:
```

Comment:
We could make this a downloadable manifest; see `code_sample` in https://kubernetes.io/docs/contribute/style/hugo-shortcodes/#source-code-files